### ISSUE TITLE
installing update of SuperTool requires restart of Gramps

### ISSUE THREAD

**Issue by emyoulation** [Taapeli/isotammi-addons#36]
I had SuperToo1 1.4.7 installed and was using it in the session. But closed it and used the Addon Manager to update to the 1.4.8 version. However, the 1.4.7 version continued to run until restarting Gramps.

I wonder if the Addon Manager needs to re-run the Plugin Registration after a successful addon update. (Which still would not allow views to update since the GUI has that data cached.  And  features using dialogs in the Windows menu would likewise be locked.)

**Comment by emyoulation** [comment]
Not an Isotammi responsibility to correct... just thought you'd want to be aware.

**Comment by kkujansuu** [comment]
This is probably because the Supertool.py module is loaded at Gramps startup. Therefore all the classes in the module remain in memory even if the code on disk is updated and the tool itself is closed.

This behaviour changed at a recent version when I introduced the "supertool_execute" function. This function requires that the Supertool code is always available.

It might be possible to fix this but maybe it's not worth the effort.

**Comment by emyoulation** [comment]
I agree that it is probably not worth effort to resolve. But publicly documenting the behavior could set expectations instead.
